### PR TITLE
Task2 selenium we

### PR DESCRIPTION
**Pull request by BorysFe**

### CODE REVIEW

File: src/main/java/com/trello/BoardPage.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public void openSearchField() {
+        searchField.click();
+    }
+
+    public void openMemberMenu() {
+        openMemberMenuButton.click();
+    }
+
+    public void newBoardTitle() {
+
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: ?

File: src/main/java/com/trello/BoardPage.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        newBoardLink.click();
+        Waiters.waitSeconds(4);
+        newBoardTitleField.clear();
+        newBoardTitleField.sendKeys(newBoardTitle);
+        newBoardSubmit.click();
+        Waiters.waitSeconds(3);
+    }
+
+    public String getClosedBoardMessage() {
+    return closeBoardMessage.getText();
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: code format

File: src/main/java/com/trello/LogInPage.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    private WebElement errorMeпssage;
+
+    @FindBy(partialLinkText = "error")
+    private WebElement errorMаessage;
+
+    @FindBy(tagName = "error")
+    private WebElement erdddrorMessage;
+
+
+    String emailUser1 = "fesenko.b@icloud.com";
```
Comment (edward-astapov) [inline src/main/java/com/trello/LogInPage.java]: Remove credentials from page. All values should be passed from test.

File: src/main/java/com/trello/LogInPage.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        PageFactory.initElements(driver, this);
+        driver.get("https://trello.com");
+        openLogInPageButton.click();
+    }
+
+    public void openLogInPage() {
+        openLogInPageButton.click();
+    }
+
+    public void logInExistedUser1() {
```
Comment (edward-astapov) [inline src/main/java/com/trello/LogInPage.java]: Need single login method that receives email and password from test

File: src/test/java/com/trello/authorization/AuthorisationTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class AuthorisationTest {
+
+    private ChromeDriver driver;
+    LogInPage logInPage = new LogInPage(driver);
```
Comment (edward-astapov) [inline src/test/java/com/trello/authorization/AuthorisationTest.java]: Do not create page object as test field. Use separate pages in each test

File: src/test/java/com/trello/boards/NewBoardTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    @BeforeClass
+    public void webDriver() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
+        driver = new ChromeDriver();
+    }
+
+    @AfterMethod
+    public void signOffFromTrello() {
+//        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
```
Comment (edward-astapov) [inline src/test/java/com/trello/boards/NewBoardTest.java]: If code is not needed - please remove it

File: src/test/java/com/trello/boards/NewBoardTest.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Test
+    public void userExistedBoardCheck() {
+        LogInPage logInPage = new LogInPage(driver);
+        BoardPage boardPage = new BoardPage(driver);
+        Actions action = new Actions(driver);
+
+        WebElement userBoard = driver.findElement(By.xpath(".//a[@title= 'Borys_Tech-Stack']"));
+        String boardTitle = driver.findElement(By.xpath(".//input[@class= 'board-name-input js-board-name-input']"))
```
Comment (edward-astapov) [inline src/test/java/com/trello/boards/NewBoardTest.java]: If page object used - it is not recommended to do findElement in test class. Move this to page object, if possible

File: src/test/java/com/trello/boards/NewBoardTest.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        LogInPage logInPage = new LogInPage(driver);
+        BoardPage boardPage = new BoardPage(driver);
+
+        String newBoardTitle = "Auto test - "+System.currentTimeMillis();
+
+        logInPage.logInExistedUser1();
+        boardPage.openMemberMenu();
+        boardPage.addNewBoard(newBoardTitle);
+
+        Assert.assertEquals(boardPage.boardTitle.getAttribute("value"), newBoardTitle, "The board title is wrong");
```
Comment (edward-astapov) [inline src/test/java/com/trello/boards/NewBoardTest.java]: Move methods like this to page classes

File: src/test/java/com/trello/registration/RegistrationTest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.asserts.Assertion;
+
+import java.util.List;
+
+public class RegistrationTest {
+
+    private ChromeDriver driver;
+    RegistrationPage registrationPage = new RegistrationPage(driver);
```
Comment (edward-astapov) [inline src/test/java/com/trello/registration/RegistrationTest.java]: Use separate pages in each test

File: src/test/java/com/trello/registration/RegistrationTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+
+    @BeforeClass
+    public void webDriver() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
+//        driver = new ChromeDriver();
```
Comment (edward-astapov) [inline src/test/java/com/trello/registration/RegistrationTest.java]: ?

File: src/test/java/com/trello/registration/RegistrationTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+    @AfterMethod
+    public void signOffFromTrello() {
+        BoardPage boardPage = new BoardPage(driver);
+        boardPage.logOutUser();
+    }
+
+    @Test
+    public void privacyPolicyTextCheck() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
```
Comment (edward-astapov) [inline src/test/java/com/trello/registration/RegistrationTest.java]: We decide to move this in @BeforeClass

File: src/main/java/com/trello/BoardPage.java [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    private WebElement newBoardLink;
+
+    @FindBy (xpath = ".//input[@placeholder= 'Add board title']")
+    private WebElement newBoardTitleField;
+
+    @FindBy (xpath = ".//button[@data-test-id='create-board-submit-button']")
+    private WebElement newBoardSubmit;
+
+    @FindBy (xpath = ".//input[@class= 'board-name-input js-board-name-input']")
+    public WebElement boardTitle;
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: ALL WebElements in Page classes should be **private**
After changing all elements to private - please update affected code, by moving some methods from test to page class (see comments in tests for more details)

**Review by edward-astapov — CHANGES_REQUESTED** [review]
Please fix comments

File: src/main/java/com/trello/BoardPage.java (hunk 12 above)
Comment (BorysFe) [reply src/main/java/com/trello/BoardPage.java]: ok

File: src/main/java/com/trello/LogInPage.java (hunk 3 above)
Comment (BorysFe) [reply src/main/java/com/trello/LogInPage.java]: ok

File: src/test/java/com/trello/boards/NewBoardTest.java (hunk 8 above)
Comment (BorysFe) [reply src/test/java/com/trello/boards/NewBoardTest.java]: Done

File: src/test/java/com/trello/boards/NewBoardTest.java (hunk 7 above)
Comment (BorysFe) [reply src/test/java/com/trello/boards/NewBoardTest.java]: Done

File: src/test/java/com/trello/boards/NewBoardTest.java (hunk 6 above)
Comment (BorysFe) [reply src/test/java/com/trello/boards/NewBoardTest.java]: Removed

File: src/test/java/com/trello/authorization/AuthorisationTest.java (hunk 5 above)
Comment (BorysFe) [reply src/test/java/com/trello/authorization/AuthorisationTest.java]: Done, if I understand right.

File: src/main/java/com/trello/LogInPage.java (hunk 4 above)
Comment (BorysFe) [reply src/main/java/com/trello/LogInPage.java]: Done

File: src/main/java/com/trello/LogInPage.java (hunk 3 above)
Comment (BorysFe) [reply src/main/java/com/trello/LogInPage.java]: Done

File: src/main/java/com/trello/BoardPage.java (hunk 2 above)
Comment (BorysFe) [reply src/main/java/com/trello/BoardPage.java]: Done

File: src/main/java/com/trello/BoardPage.java [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+    public String getBoardTitleAttribute(String attributeName) {
+        return  boardTitle.getAttribute(attributeName);
+    }
+
+    public String getDeletedMessage() {
+        return deleteBoardMessage.getText();
+    }
+
+    public String getCustomUserBoardTitle(String eee) {
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: 'eee' is not valid  variable name

File: src/main/java/com/trello/BoardPage.java [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+    @FindBy (xpath = ".//a[@class='quiet js-delete']")
+    private   WebElement deleteBoard;
+
+    @FindBy (xpath = ".//input[@value='Delete']")
+    private WebElement confirmationDeleteBoardBtn;
+
+    @FindBy (xpath = ".//div[@class= 'js-react-root']//h1")
+    private WebElement deleteBoardMessage;
+
+    private final String userBoard = (".//a[@title= '%s']");
```
Comment (edward-astapov) [inline src/main/java/com/trello/BoardPage.java]: You can remove brackets

File: src/main/java/com/trello/LogInPage.java [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+    @FindBy(id = "login")
+    private WebElement logInButton;
+
+    @FindBy(id = "error")
+    private WebElement errorMessage;
+
+    public LogInPage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        driver.get("https://trello.com");
```
Comment (edward-astapov) [inline src/main/java/com/trello/LogInPage.java]: Please avoid this in future.

File: src/main/java/com/trello/RegistrationPage.java [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+    @FindBy(xpath = "//p[@class= 'error-message']")
+    private WebElement errorMessage;
+
+    @FindBy(xpath = ".//button[@data-test-id= 'moonshot-try-bc-free-trial']")
+    private WebElement trialBtn;
+
+    public RegistrationPage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        driver.get("https://trello.com");
```
Comment (edward-astapov) [inline src/main/java/com/trello/RegistrationPage.java]: Same here. Avoid extra code in constructors

File: src/test/java/com/trello/authorization/AuthorisationTest.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)" +
+                "//chromedriver87.exe");
+        driver = new ChromeDriver();
+    }
+
+    @AfterMethod
+    public void signOffFromTrello() {
+        BoardPage boardPage = new BoardPage(driver);
+        boardPage.logOutIfAuthorised();
+        ;
```
Comment (edward-astapov) [inline src/test/java/com/trello/authorization/AuthorisationTest.java]: ?

File: src/test/java/com/trello/authorization/AuthorisationTest.java [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+import org.testng.annotations.Test;
+
+public class AuthorisationTest {
+
+    private ChromeDriver driver;
+
+    @BeforeClass
+    public void webDriver() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)" +
+                "//chromedriver87.exe");
```
Comment (edward-astapov) [inline src/test/java/com/trello/authorization/AuthorisationTest.java:21]: Avoid line breaks like this

**Review by edward-astapov — CHANGES_REQUESTED** [review]
Need works

**Review by edward-astapov — APPROVED** [review]

**Pull request merged**